test/acceptance/features/mockhttp.feature

```diff
@@ -0,0 +1,118 @@
+      And the HTTP response body must have the JSON properties
+          | value | test mock response |
+
+  @mockhttp
+  Scenario: Mock permanent request
+    Given I store "[UUID]" in context "id"
+      And I mock the HTTP request at "[CONF:httpMockUrl]" with the JSON
+      """
+      {
+        "permanent": true,
+        "request": {
+          "method": "DELETE",
+          "path": "/test/[CTXT:id]"
+        },
+        "response": {
+          "status": 204,
+          "body": ""
+        }
+      }
+      """
+    Given the HTTP endpoint "[CONF:httpMockUrl]/test/[CTXT:id]"
+     When I send a HTTP "DELETE" request
+     Then the HTTP status code must be "204"
+      And the HTTP response body must be empty
+     When I send a HTTP "DELETE" request
+     Then the HTTP status code must be "204"
+      And the HTTP response body must be empty
+
+  @mockhttp
+  Scenario: Mock one-shot request
+    Given I store "[UUID]" in context "id"
+      And I mock the HTTP request at "[CONF:httpMockUrl]" with the JSON
+      """
+      {
+        "permanent": false,
+        "request": {
+          "method": "DELETE",
+          "path": "/test/[CTXT:id]"
+        },
+        "response": {
+          "status": 204,
+          "body": ""
+        }
+      }
+      """
+    Given the HTTP endpoint "[CONF:httpMockUrl]/test/[CTXT:id]"
+     When I send a HTTP "DELETE" request
+     Then the HTTP status code must be "204"
+      And the HTTP response body must be empty
+     When I send a HTTP "DELETE" request
+     Then the HTTP status code must be "404"
+
+  @mockhttp
+  Scenario: Mock request with full configuration
+    Given I store "[UUID]" in context "id"
+      And I mock the HTTP request at "[CONF:httpMockUrl]" with the JSON
+      """
+      {
+        "request": {
+          "method": "POST",
+          "path": "/test/[CTXT:id]"
+        },
+        "response": {
+          "status": 201,
+          "headers": {
+            "Content-Type": ["application/json"]
+          },
+          "body": "{\"value\": \"test mock response\"}"
+        }
+      }
+      """
+    Given the HTTP endpoint "[CONF:httpMockUrl]/test/[CTXT:id]"
+     When I send a HTTP "POST" request
+     Then the HTTP status code must be "201"
+      And the HTTP request headers
+          | Content-Type | application/json |
+      And the HTTP response body must have the JSON properties
+          | value | test mock response |
+
+  @mockhttp
+  Scenario: Mock request with timeout
+    Given I store "[UUID]" in context "id"
+      And I mock the HTTP request at "[CONF:httpMockUrl]" with the JSON
+      """
+      {
+        "request": {
+          "method": "GET",
+          "path": "/test/[CTXT:id]"
+        },
+        "latency": 1000
+      }
+      """
+    Given the HTTP endpoint "[CONF:httpMockUrl]/test/[CTXT:id]"
+      And an HTTP timeout of "300" milliseconds
+     When I send a HTTP "GET" request
+     Then the HTTP response timed out
+
```

Comment:
I missed an scenario to test timeout with latency negative (so that, you do not specify the timeout for the caller to be less than the latency)

Reply:
I've removed the possibility of using a negative latency